website/src/content/docs/docs/libraries/http/operations.md

```diff
@@ -580,6 +580,150 @@ model Thing {
 }
 ```
 
+## JSON Merge-Patch
+
+[RFC 7396](https://www.rfc-editor.org/rfc/rfc7396) describes a standard for interpreting
+a Patch request body using content-type `application/merge-patch+json`to update an existing
+resource. The TypeSpec Http library provides Model templates `MergePatchUpdate` and
+`MergePatchCreateOrUpdate` for specifying a JSON merge-patch request body. The templates
+recursively apply the merge-patch transformation rules to a TypeSpec resource Model, taking
+into account the structure and characteristics of the resource Model.
+
+For example, given a resource model like this:
+
+```tsp
+model Resource {
+  id: string;
+  name?: string;
+  quantity?: safeint;
+  color: "blue" | "green" | "red" = "blue";
+  flavor?: "vanilla" | "chocolate" | "strawberry" = "vanilla";
+  related?: Record<Resource>;
+  tags?: string[];
+}
+```
+
+A JSON Merge-Patch request updating this resource would have the following behavior:
+
+- `id` may or may not occur in the request, if it does occur, the resource value can be updated to a new string, but cannot be erased by sending null.
+- `name` may or may not occur in the request, if it does occur, the resource value can be updated to a new string or erased by sending null.
+- `quantity` may or ay not occur in the request, if it does occur, the resource value can be updated to a new integer value or erased by sending null.
+- `color` may or may not occur in the request, if it does occur, the resource value can be updated to one of the appropriate values. If set to `null` it is returned to its default value (`blue`)
+- `flavor` may or may not occur in the request, if it does occur, the resource value can be updated to one of the appropriate values. If set to `null` it is returned to its default value (`vanilla`)
+- `related` may or may not occur in the request, if it does occur, the resource value can be updated or erased (set to `{}`) by sending null.
```

Comment:
"Erased" in this context (`related`) should set the value of the resource so that it has _no_ `related` record, not an empty record, right? Are records special in this respect? It seems like if you had a resource with `related: { ... }` and you wanted to set it to `related: {}`, you would have to first send `related: null` and then `related: {}`.